module/attribute/src/Application/Validator/AttributeCodeValidator.php

```diff
@@ -50,8 +51,12 @@ public function validate($value, Constraint $constraint): void
             return;
         }
 
-        if (!preg_match($constraint->pattern, $value)) {
+        if (!preg_match($constraint->pattern, $value) || 0 === strpos(
+            $value,
+            SystemAttributeCode::SYSTEM_ATTRIBUTE_PREFIX
```

Comment:
Can't we extend AttributeCode value object to deny possibility of creating with `esa_` prefix?